pkg/ingress/config/config.go

```diff
@@ -39,31 +40,50 @@ type kubeConfig struct {
 
 // OpenStack credentials configuration, the section is required.
 type osConfig struct {
-	Username  string
-	Password  string
-	ProjectID string `mapstructure:"project_id"`
-	AuthURL   string `mapstructure:"auth_url"`
-	Region    string
+	Username   string
+	UserID     string `mapstructure:"user_id"`
+	TrustID    string `mapstructure:"trust_id"`
+	Password   string
+	ProjectID  string `mapstructure:"project_id"`
+	AuthURL    string `mapstructure:"auth_url"`
+	Region     string
+	DomainID   string `mapstructure:"domain_id"`
+	DomainName string `mapstructure:"domain_name"`
+	CAFile     string `mapstructure:"ca_file"`
 }
 
 // Octavia service related configuration
 type octaviaConfig struct {
 	// (Required)Subnet ID to create the load balancer.
 	SubnetID string `mapstructure:"subnet_id"`
 
-	// (Optional)Public network to create floating IP.
+	// (Optional)Public network ID to create floating IP.
 	// If empty, no floating IP will be allocated to the load balancer vip.
-	FloatingIPNetwork string `mapstructure:"fip_network"`
+	FloatingIPNetwork string `mapstructure:"floating_network_id"`
 }
 
 // ToAuthOptions gets openstack auth options
 func (cfg Config) ToAuthOptions() gophercloud.AuthOptions {
 	return gophercloud.AuthOptions{
 		IdentityEndpoint: cfg.OpenStack.AuthURL,
 		Username:         cfg.OpenStack.Username,
+		UserID:           cfg.OpenStack.UserID,
 		Password:         cfg.OpenStack.Password,
 		TenantID:         cfg.OpenStack.ProjectID,
-		DomainName:       "default",
+		DomainID:         cfg.OpenStack.DomainID,
+		DomainName:       cfg.OpenStack.DomainName,
+		AllowReauth:      true,
+	}
+}
+
+func (cfg Config) ToV3AuthOptions() tokens.AuthOptions {
```

Comment:
Openstack now supports only v3 right, so maintaining 2 configs is necessary? ToAuthOptions & ToV3AuthOptions? could you let me know if there is any guideline that cloud-provider-openstack maintains compatibility upto this release of openstack?

Reply:
@lingxiankong yes, its been removed in queens release. https://docs.openstack.org/releasenotes/keystone/queens.html#other-notes

Reply:
Thanks for the link, so yeah, we don't need to maintain that. But I'd rather do that in a separate PR, thoughts? BTW, just curious, are you using octavia-ingress-controller?